cmd/state-svc/main.go

```diff
@@ -6,49 +6,136 @@ import (
 	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/constants"
 	"github.com/ActiveState/cli/internal/errs"
 	"github.com/ActiveState/cli/internal/logging"
 )
 
+type command string
+
+const (
+	CmdStart      = "start"
+	CmdStop       = "stop"
+	CmdStatus     = "status"
+	CmdForeground = "foreground"
+)
+
+var commands = []command{
+	CmdStart,
+	CmdStop,
+	CmdStatus,
+	CmdForeground,
+}
+
 func main() {
-	if err := run(); err != nil {
-		fmt.Fprintln(os.Stderr, errs.Join(err, ": ").Error())
+	if os.Getenv("VERBOSE") == "true" {
+		logging.CurrentHandler().SetVerbose(true)
+	}
+
+	err := run()
+	if err != nil {
+		errMsg := errs.Join(err, ": ").Error()
+		logging.Errorf("state-svc errored out: %s", errMsg)
+		fmt.Fprintln(os.Stderr, errMsg)
 		os.Exit(1)
 	}
 }
 
 func run() error {
-	s, err := server.New()
-	if err != nil {
-		return errs.Wrap(err, "Could not create server")
+	var cmd command = ""
+	if len(os.Args) > 1 {
+		cmd = command(os.Args[1])
 	}
 
 	cfg, err := config.New()
 	if err != nil {
 		return errs.Wrap(err, "Could not initialize config")
 	}
-	if err := cfg.Set("port", s.Port()); err != nil {
-		return errs.Wrap(err, "Could not save config")
+
+	switch cmd {
+	case CmdStart:
+		logging.Debug("Running CmdStart")
+		return runStart(cfg)
+	case CmdStop:
+		logging.Debug("Running CmdStop")
+		return runStop(cfg)
+	case CmdStatus:
+		logging.Debug("Running CmdStatus")
+		return runStatus(cfg)
+	case CmdForeground:
+		logging.Debug("Running CmdForeground")
+		return runForeground(cfg)
 	}
 
+	return errs.New("Expected one of following commands: %v", commands)
+}
+
+func runForeground(cfg *config.Instance) error {
+	logging.Debug("Running in Foreground")
+
+	p := NewService(cfg)
+
 	// Handle sigterm
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt, syscall.SIGTERM)
+	defer signal.Stop(sig)
 
 	go func() {
-		oscall := <-c
+		oscall := <-sig
 		logging.Debug("system call:%+v", oscall)
-		if err := s.Close(); err != nil {
-			fmt.Fprintf(os.Stderr, "Closing server failed: %v", err)
+		if err := p.Stop(); err != nil {
+			logging.Error("Stop on sigterm failed: %v", errs.Join(err, ": "))
 		}
-		signal.Stop(c)
 	}()
 
-	if err := s.Start(); err != nil {
-		return errs.Wrap(err, "Failed to start server")
+	if err := p.Start(); err != nil {
+		return errs.Wrap(err, "Could not start service")
+	}
+
+	return nil
+}
+
+func runStart(cfg *config.Instance) error {
+	logging.Debug("Starting")
+
+	s := NewServiceManager(cfg)
+	if err := s.Start(os.Args[0], CmdForeground); err != nil {
+		return errs.Wrap(err, "Could not start serviceManager")
+	}
+
+	return nil
+}
+
+func runStop(cfg *config.Instance) error {
+	logging.Debug("Starting")
+
+	s := NewServiceManager(cfg)
+	if err := s.Stop(); err != nil {
+		return errs.Wrap(err, "Could not stop serviceManager")
+	}
+
+	return nil
+}
```

Comment:
nit: I think these debugs are redundant. If not, then the runStop debug message should be altered, and other run functions might need to have debugs added.